logstreams/src/main/java/io/camunda/zeebe/logstreams/log/LogStreamBatchReader.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.logstreams.log;
+
+import io.camunda.zeebe.logstreams.log.LogStreamBatchReader.Batch;
+import io.camunda.zeebe.util.CloseableSilently;
+import java.util.Iterator;
+
+/**
+ * Reads the log stream in batches. Similar to {@link LogStreamReader} but groups events with the
+ * same source event position in batches. Can be used to read all follow-up events atomically.
```

Comment:
```suggestion
 * same source event position in batches. Can be used to read all follow-up events at once.
```

Idk but atomically sounds wrong in this context.